creator-node/src/middlewares.js

```diff
@@ -120,30 +119,28 @@ async function triggerSecondarySyncs (req) {
 }
 
 /** Retrieves current FQDN registered on-chain with node's owner wallet. */
+// TODO - this can all be cached on startup, but we can't validate the spId on startup unless the
+// services has been registered, and we can't register the service unless the service starts up.
+// Bit of a chicken and egg problem here with timing of first time setup, but potential optimization here
 async function getOwnEndpoint (req) {
   if (config.get('isUserMetadataNode')) throw new Error('Not available for userMetadataNode')
   const libs = req.app.get('audiusLibs')
 
-  let spOwnerWallet
-  if (config.get('spOwnerWallet')) {
-    spOwnerWallet = config.get('spOwnerWallet')
-  } else if (config.get('ethWallets') && config.get('spOwnerWalletIndex') && Array.isArray(config.get('ethWallets')) && config.get('ethWallets').length > config.get('spOwnerWalletIndex')) {
-    spOwnerWallet = config.get('ethWallets')[config.get('spOwnerWalletIndex')]
-  } else {
-    throw new Error('Must provide either spOwnerWallet or ethWallets and spOwnerWalletIndex config vars.')
-  }
-
-  const spId = await libs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromAddress(spOwnerWallet, 'creator-node')
-  const spInfo = [await libs.ethContracts.ServiceProviderFactoryClient.getServiceProviderInfo('creator-node', spId)]
+  let creatorNodeEndpoint = config.get('creatorNodeEndpoint')
+  if (!creatorNodeEndpoint) throw new Error('Must provide either creatorNodeEndpoint config var.')
 
+  const spId = await libs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(creatorNodeEndpoint)
+  if (!spId) throw new Error('Cannot get spId for node')
```

Comment:
will this prevent startup for an unregistered node or just prevent operations from going through? if its the latter we are perfect

Reply:
it will prevent ops from going through. it'll startup just fine